hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -288,9 +288,6 @@ private HoodieWriteConfig createMetadataWriteConfig(
         .withCompactionConfig(HoodieCompactionConfig.newBuilder()
             .withInlineCompaction(false)
             .withMaxNumDeltaCommitsBeforeCompaction(writeConfig.getMetadataCompactDeltaCommitMax())
-            // by default, the HFile does not keep the metadata fields, set up as false
-            // to always use the metadata of the new record.
-            .withPreserveCommitMetadata(false)
```

Comment:
Have you audited the code path to see there's no behavior change for metadata writing?

Reply:
What is the background to bring back the metadata fields for MDT?

Reply:
sorry, default value for preserveCommitMetadata is true. but for hfile(MDT), meta fields are disable and hence we need to re-create and not preserve. can you help me clarify.

Reply:
@lokeshj1703 This is ok, though we actually don't need the differentiating logic for metadata table and it can be set to `true`, given that MDT does not have the meta fields and there is no difference in logic between setting it to `true` and `false`.  I'm going to land this PR now.  Let's put up a separate PR to clean it up.

Reply:
@yihua created https://github.com/apache/hudi/pull/8393